app/agents/voice/automatic/__init__.py

```
@@ -347,9 +368,23 @@ async def run_pipeline():
             root_span.set_attribute("user.name", user_name)
             root_span.set_attribute("service.name", "breeze-voice-agent")
-            langfuse_client.update_current_trace(user_id=user_name)
-            langfuse_client.update_current_trace(session_id=args.session_id)
-            langfuse_client.update_current_trace(tags=[voice_name])
+
+            # Optional client session attributes (from release)
+            if getattr(args, "client_sid", None):
+                root_span.set_attribute("client_sid", args.client_sid)
+                root_span.set_attribute(
+                    "application_logs",
+                    generate_open_observer_url_for_session_id(args.client_sid)
```

Ruff (0.12.2)

Undefined name `generate_open_observer_url_for_session_id`

(F821)